tracer/src/Datadog.Trace/OTelMetrics/MetricState.cs

```diff
@@ -91,39 +97,47 @@ public void RecordMeasurementDouble(double value, ReadOnlySpan<KeyValuePair<stri
         }
     }
 
-    public void RecordMeasurementGaugeLong(long value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-        => GetOrCreatePoint(tags).UpdateGauge(value);
-
-    public void RecordMeasurementGaugeDouble(double value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-        => GetOrCreatePoint(tags).UpdateGauge(value);
-
-    public void RecordMeasurementHistogramLong(long value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-        => GetOrCreatePoint(tags).UpdateHistogram(value);
-
-    public void RecordMeasurementHistogramDouble(double value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-        => GetOrCreatePoint(tags).UpdateHistogram(value);
-
-    public IEnumerable<MetricPoint> GetMetricPoints() => _points.Values;
-
-    public MetricStreamIdentity GetIdentity() => _identity;
+    /// <summary>
+    /// Builds metric point snapshots and adds them to the provided list.
+    /// For Delta temporality, resets the running values after snapshotting.
+    /// </summary>
+    public void BuildPoints(List<MetricPoint> into)
+    {
+        foreach (var point in _points.Values)
+        {
+            var snapshot = point.CreateSnapshotAndReset();
+            into.Add(snapshot);
+        }
+    }
 
     private MetricPoint GetOrCreatePoint(ReadOnlySpan<KeyValuePair<string, object?>> tags)
     {
         var tagSet = TagSet.FromSpan(tags);
 
-        var tagDict = new Dictionary<string, object?>();
+        if (_points.TryGetValue(tagSet, out var existingPoint))
+        {
+            return existingPoint;
+        }
+
+        var dict = new Dictionary<string, object?>(tags.Length);
         for (int i = 0; i < tags.Length; i++)
         {
-            var kv = tags[i];
-            tagDict[kv.Key] = kv.Value;
+            ref readonly var kv = ref tags[i];
```

Comment:
This looks strange to me what was the rational to go to this from?

Not saying that this is incorrect as I haven't tested this out or looked _entirely_ at the complete call chain


```suggestion
            var kv = tags[i];
```

Reply:
Not really needed ATM just AI things that claim performance.